From 47d502353720004fa2d0e7b0065994b75b3e0ded Mon Sep 17 00:00:00 2001
From: Nate Graham <nate@kde.org>
Date: Wed, 2 Apr 2025 20:57:09 +0000
Subject: [PATCH] applets/notifications: improve paddings again

Zero top padding is fine when there's only a summary label, but looks
bad when there's body text, an icon, or both. Correct that.


(cherry picked from commit 53b6289f84094d91745b79c3bc902e7f80f235ef)

Co-authored-by: Nate Graham <nate@kde.org>
---
 .../package/contents/ui/delegates/DelegatePopup.qml           | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/applets/notifications/package/contents/ui/delegates/DelegatePopup.qml b/applets/notifications/package/contents/ui/delegates/DelegatePopup.qml
index 2038f49de50..e22a8322196 100644
--- a/applets/notifications/package/contents/ui/delegates/DelegatePopup.qml
+++ b/applets/notifications/package/contents/ui/delegates/DelegatePopup.qml
@@ -93,6 +93,8 @@ BaseDelegate {
 
     Components.Summary {
         id: summary
+        // Base layout intentionally has no row spacing, so add top padding here when needed
+        Layout.topMargin: delegateRoot.hasBodyText || icon.visible ? Kirigami.Units.smallSpacing : 0
         Layout.fillWidth: true
         Layout.row: 2
         Layout.column: delegateRoot.__firstColumn
@@ -108,7 +110,7 @@ BaseDelegate {
 
     Components.Icon {
         id: icon
-        // We removed the row spacing from the base layout, so re-add it just here
+        // Base layout intentionally has no row spacing, so add top padding here
         Layout.topMargin: Kirigami.Units.smallSpacing
         Layout.row: 2
         Layout.column: delegateRoot.__firstColumn + 1
-- 
GitLab

